src/UCommerce.SiteCore.Installer/Steps/SitecoreWebconfigMerger.cs

```diff
@@ -1,34 +1,50 @@
+            _loggingService = loggingService;
+            _baseDirectory = baseDirectory;
         }
-        public void Run(ITaskOutput output, NameValueCollection metaData)
+
+        public async Task Run()
         {
-            var mergeConfig = new MergeConfig(
-                "~/web.config",
-                new List<Transformation>(){
-                    new Transformation("~/sitecore modules/Shell/ucommerce/install/CleanConfig.config"),
-                    new Transformation("~/sitecore modules/Shell/ucommerce/install/uCommerce.config"),
-                    new Transformation("~/sitecore modules/Shell/ucommerce/install/uCommerce.IIS7.config", isIntegrated: true),
-                    new Transformation("~/sitecore modules/Shell/ucommerce/install/uCommerce.dependencies.sitecore.config"),
-                    new Transformation("~/sitecore modules/Shell/ucommerce/install/sitecore.config"),
-                    new Transformation("~/sitecore modules/Shell/ucommerce/install/ClientDependency.config"),
-                    new Transformation("~/sitecore modules/Shell/ucommerce/install/updateAssemblyBinding.config")
+            _loggingService.Information<SitecoreWebconfigMerger>("Merging sitecore and ucommerce config files");
+            var mergeConfig = new MergeConfig(Path.Combine(_baseDirectory.FullName, "web.config"),
+                new List<Transformation>
+                {
+                    new Transformation(Path.Combine(_baseDirectory.FullName, "sitecore modules", "Shell", "ucommerce", "install", "CleanConfig.config")),
+
+                    new Transformation(Path.Combine(_baseDirectory.FullName, "sitecore modules", "Shell", "ucommerce", "install", "uCommerce.config")),
+                    new Transformation(Path.Combine(_baseDirectory.FullName, "sitecore modules", "Shell", "ucommerce", "install", "uCommerce.IIS7.config"),
+                        isIntegrated: true),
+                    new Transformation(Path.Combine(_baseDirectory.FullName,
+                        "sitecore modules",
+                        "Shell",
+                        "ucommerce",
+                        "install",
+                        "uCommerce.dependencies.sitecore.config")),
+                    new Transformation(Path.Combine(_baseDirectory.FullName, "sitecore modules", "Shell", "ucommerce", "install", "sitecore.config")),
+                    new Transformation(Path.Combine(_baseDirectory.FullName, "sitecore modules", "Shell", "ucommerce", "install", "ClientDependency.config")),
+                    new Transformation(Path.Combine(_baseDirectory.FullName,
+                        "sitecore modules",
+                        "Shell",
+                        "ucommerce",
+                        "install",
+                        "updateAssemblyBinding.config"))
                 }
             );
 
-            mergeConfig.Run(output, metaData);
+            await mergeConfig.Run();

```

Comment:
How can you await a method that isn't async?

Reply:
Good point